doc/src/learn/cryptography/sui-move-signatures.md

```diff
@@ -0,0 +1,167 @@
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme ed25519 --msg $MSG --secret-key  $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's verify method. All inputs are represented in bytes in hex format:
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let verify = ed25519::ed25519_verify(&sig, &pk, &msg);
+    assert!(verify == true, 0);
+```
+
+ 1. Secp256k1 non-recoverable signature (64 bytes)
+
+```shell
+target/debug/sigs-cli keygen --scheme secp256k1 --seed 0000000000000000000000000000000000000000000000000000000000000000                
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme secp256k1 --msg $MSG --secret-key $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's verify method:
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let verify = ecdsa_k1::secp256k1_verify(&sig, &pk, &msg);
+    assert!(verify == true, 0);
+```
+
+ 1. Secp256k1 recoverable signature (65 bytes)
+
+```shell
+target/debug/sigs-cli keygen --scheme secp256k1-rec --seed 0000000000000000000000000000000000000000000000000000000000000000                
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme secp256k1-rec --msg $MSG --secret-key $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's ecrecover method and check equality:
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let recovered = ecdsa_k1::secp256k1_ecrecover(&sig, &msg);
+    assert!(pk == recovered, 0);
+```
+
+ 1. Secp256r1 non-recoverable signature (64 bytes)
+
+```shell
+target/debug/sigs-cli keygen --scheme secp256r1 --seed 0000000000000000000000000000000000000000000000000000000000000000                
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme secp256r1 --msg $MSG --secret-key $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's verify method:
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let verify = ecdsa_r1::secp256r1_verify(&sig, &pk, &msg);
+    assert!(verify == true, 0);
+```
+
+ 1. Secp256r1 recoverable signature (65 bytes)
+
+```shell
+target/debug/sigs-cli keygen --scheme secp256r1-rec --seed 0000000000000000000000000000000000000000000000000000000000000000                
+Private key in hex: $SK
+Public key in hex: $PK
+
+target/debug/sigs-cli sign --scheme secp256r1-rec --msg $MSG --secret-key $SK
+
+Signature in hex: $SIG
+Public key in hex: $PK
+```
+
+Call smart contract's ecrecover method and check equality:
+
+```move
+    let msg = x"$MSG";
+    let pk = x"$PK";
+    let sig = x"$SIG";
+    let recovered = ecdsa_r1::secp256r1_ecrecover(&sig, &msg);
+    assert!(pk == recovered, 0);
+```
+
+ 1. BLS G1 signature (minSig setting)
```

Comment:
are we going to add example for minPK setting as well?

Reply:
yeah need to merge fastcrypto https://github.com/MystenLabs/fastcrypto/pull/579 first